### ISSUE TITLE
Dark Mode?

### ISSUE THREAD

**Issue by perguto** [crow-translate/crow-translate#487]
Is there a dark mode for the Windows GUI? I didn't find one in the settings. It would be nice to have one. Thanks for this awesome program btw!

**Comment by VolkMilit** [comment]
It depends on your system settings. Windows 10 has an option to turn dark mode.

**Comment by Shatur** [comment]
Icon support is needed. We need to bundle both variants and select automatically the correct one based on system settings.

**Comment by VolkMilit** [comment]
Qt 5 doesn't have an native API for dark\lite theme, or I'm not aware where to look. On windows we can use register to determine user settings in `HKEY_CURRENT_USER\\SOFTWARE\\Microsoft\\Windows\\CurrentVersion\\Themes\\Personalize`

**Comment by JIGASHI** [comment]
Could you make this a priority.

**Comment by vvkdev** [comment]
How to activate the dark mode? My Windows 10 dark mode is on and other apps are dark, but Crow not.

**Comment by VolkMilit** [comment]
@vvkdev AFAIK, you can't, not with Qt 5 anyway. Qt 6 support this natively, but some Linux distribution still ship old Qt 5, that's why we wait for them.

**Comment by AlbertoFabbri93** [comment]
I installed the flatpak version of Crow Translate on my Linux system (OpenSuse Aeon with Gnome) and when I enable dark mode this is how it looks:

![Screenshot from 2023-07-31 00-40-39](https://github.com/crow-translate/crow-translate/assets/51910894/be20c458-4b53-4843-bb61-a7de819ab794)

It is impossible to see the icons of the buttons.

**Comment by MesterPerfect** [comment]
Before, I used obs-studio on Windows 32-bit, which means it was utilizing Qt5. The program's default theme was dark. Therefore, I believe achieving a dark theme is possible with either Qt5 or Qt6.

**Comment by Shatur** [comment]
OBS uses custom styles, it's not a default dark Qt theme.

**Comment by Gentleman03** [comment]
> OBS uses custom styles, it's not a default dark Qt theme.

Then maybe you would be able to add some way to use custom styles?

**Comment by Shatur** [comment]
I personally dislike custom styles. It's better to just migrate to Qt6.

**Comment by Shatur** [comment]
I wish I could have more free time...
I currently quite busy with another project, so a PR would be highly appreciated. I will be able to review it quickly, just don't have enough time to sit and do the migration myself.

**Comment by MesterPerfect** [comment]
> I wish I could have more free time... I currently quite busy with another project, so a PR would be highly appreciated. I will be able to review it quickly, just don't have enough time to sit and do the migration myself.

I was hoping I could assist you, but I primarily develop using Python and have no experience with C++.

**Comment by Shatur** [comment]
It's okay, maybe some else will.

**Comment by Gentleman03** [comment]
> I personally dislike custom styles. It's better to just migrate to Qt6.

Ofcourse, especially if adding a custom styles feature isn't a quick thing to add.


> It's okay, maybe some else will.

I'm afraid it's not going to happen. Latest feature request for dark theme is 2 years old.

**Comment by Shatur** [comment]
> Latest feature request for dark theme is 2 years old.

True, but Qt6 wasn't ready for quite some time too. For example, QtMultimedia was re-introduced only in 6.2.

**Comment by Aqa-Ib** [comment]
> I'm afraid it's not going to happen. Latest feature request for dark theme is 2 years old.

This program is very useful. There are odds of someone with the knowledge to port it to Qt6. I think that a pinned issue asking for help would improve the chances. Regards.
